src/targets/gpu/write_literals.cpp

```diff
@@ -27,36 +27,142 @@
 #include <migraphx/instruction.hpp>
 #include <migraphx/program.hpp>
 #include <migraphx/env.hpp>
+#include <migraphx/liveness.hpp>
+#include <migraphx/register_op.hpp>
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace gpu {
 
 MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_COPY_LITERALS)
 
+struct literal_as_argument
+{
+    std::string name() const { return "gpu::literal_as_argument"; }
+    argument data;
+    std::string data_name = "{ ... }";
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.data_name, "data"));
```

Comment:
You arent serializing the data.

Reply:
Added back in the serialization of data, however it does print out the entire data argument which makes it very difficult to read the printout of the instructions. Tried to overwrite the printout with the following to no avail:
```cpp
friend std::ostream& operator<<(std::ostream& os, const operation& op) { return os << "testing..."; }
```

Reply:
Fixed, thanks